include/fiction/algorithms/simulation_sidb/time_to_solution.hpp

```diff
@@ -0,0 +1,118 @@
+//
+// Created by Jan Drewniok on 23.12.22.
+//
+
+#ifndef FICTION_TIME_TO_SOLUTION_HPP
+#define FICTION_TIME_TO_SOLUTION_HPP
+
+#include "fiction/algorithms//simulation_sidb/check_groundstate.hpp"
+#include "fiction/algorithms//simulation_sidb/minimum_energy.hpp"
+#include "fiction/algorithms/simulation_sidb/exhaustive_ground_state_simulation.hpp"
+#include "fiction/algorithms/simulation_sidb/quicksim.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/traits.hpp"
+
+#include <fmt/format.h>
+
+#include <algorithm>
+#include <chrono>
+#include <cstdint>
+#include <iostream>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * This struct stores the time-to-solution, the simulation accuracy and the average single simulation runtime of
+ * quicksim (see quicksim.hpp).
+ *
+ */
+struct time_to_solution_stats
+{
+    double time_to_solution{};
+    double acc{};
+    double mean_single_runtime{};
+
+    void report(std::ostream& out = std::cout)
+    {
+        out << fmt::format("time_to_solution: {} | acc: {} | t_(s): {} \n", time_to_solution, acc, mean_single_runtime);
+    }
+};
+
+/**
+ * This function determines the time-to-solution (time_to_solution) and the accuracy (acc) of the quicksim-algorithm.
+ *
+ * @paramt Lyt cell-level layout.
+ * @param charge_distribution_surface<Lyt> charge layout that is used for the simulation.
+ * @param tts_stats struct where the results (time_to_solution, acc, single runtime) are stored.
+ * @param repetitions number of repetitions to determine the simulation accuracy (repetitions = 100 ==> accuracy is
+ * precise to 1 %).
+ * @param iteration_steps simulation parameter (see quicksim.hpp).
+ * @param alpha simulation parameter (see quicksim.hpp).
+ * @param convlevel the time-to-solution also depends one the given confidence level which can be set here.
+ */
+template <typename Lyt>
+void sim_acc_tts(charge_distribution_surface<Lyt>& lyt, exgs_stats<Lyt>& result_exact,
```

Comment:
Why is this function calling `quicksim` itself but requires the user to pass `exgs_stats` from a previous run? Can it not call that itself too?